epilepsy12/tasks.py

```diff
@@ -69,10 +73,288 @@ def async_insert_old_pt_data(csv_path: str = "data.csv"):
 def async_insert_user_data(csv_path: str = "data.csv"):
     insert_user_data(csv_path=csv_path)
 
+
 @shared_task
 def hello():
     """
     THIS IS A SCHEDULED TASK THAT IS CALLED AT 06:00 EVERY DAY
     THE CRON DATE/FREQUENCY IS SET IN SETTING.PY
     """
     logger.debug("0600 cron check task ran successfully")
+
+
+@shared_task
+def download_kpi_summary_as_csv(cohort):
+    """
+    Asynchronous task to pull data from KPIAggregation tables and store as dataframe for export as CSV
+    Accepts cohort as optional param, defaults to 6
+    Output - 8 sheets of .xlsx structured as follows:
+    - Country level
+    - HBT level
+    - ICB level
+    - NHSregion_level
+    - Network_level
+    - National_level
+    - Reference
+    - National_comparison
+    """
+
+    # Define KPI measures for extraction
+    measures = [
+        "paediatrician_with_expertise_in_epilepsies",
+        "epilepsy_specialist_nurse",
+        "tertiary_input",
+        "epilepsy_surgery_referral",
+        "ecg",
+        "mri",
+        "assessment_of_mental_health_issues",
+        "mental_health_support",
+        "sodium_valproate",
+        "comprehensive_care_planning_agreement",
+        "comprehensive_care_planning_content",
+        "school_individual_healthcare_plan",
+    ]
+
+    def create_dataframe(KPI_model1, constants_list, KPI_model2=None, constants_list2=None, is_regional=False):
```

Comment:
Would it make sense to move this function out to be at the top level somewhere?

At the moment it's hard to see at a glance where it ends, although there is the `# COUNTRY - SHEET 1` comment

Reply:
Sure Michael, I think the best place for this then would be with the other aggregation functions in aggregate_by.py . That should tidy things up a bit.

Reply:
There is a folder for functions like this called `general_functions`
Might sit well there.